Assets/VRTK/Source/Scripts/Utilities/VRTK_SharedMethods.cs

```diff
@@ -634,5 +614,41 @@ public static BuildTargetGroup[] GetValidBuildTargetGroups()
             }).ToArray();
         }
 #endif
+
+        /// <summary>
+        /// The FindEvenInactiveComponentsInLoadedScenes method searches active and inactive game objects in all
+        /// loaded scenes for components matching the type supplied. 
+        /// </summary>
+        /// <param name="stopOnMatch">If true, will stop searching objects as soon as a match is found.</param>
+        /// <returns></returns>
+        private static IEnumerable<T> FindEvenInactiveComponentsInLoadedScenes<T>(bool stopOnMatch = false) where T : Component
+        {
+            List<T> results = new List<T>();
+            for (int sceneIndex = 0; sceneIndex < SceneManager.sceneCount; sceneIndex++)
+            {
+                Scene scene = SceneManager.GetSceneAt(sceneIndex);
+                if (scene.isLoaded)
+                {
+                    GameObject[] rootObjects = scene.GetRootGameObjects();
+                    foreach (GameObject rootObject in scene.GetRootGameObjects())
+                    {
+                        if (stopOnMatch)
```

Comment:
The best thing I could come up with to avoid unnecessary iteration or extra array elements when looking for a single element. 

This is the main change from @peaj solution and lets us share this method between both `FindEvenInactiveComponents` and `FindEvenInactiveComponent`